src/main/webapp/components/directives/cache/general/generaldist.js

```diff
@@ -0,0 +1,49 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.generaldist', ['ispn.services.utils']);
+
+  module.directive('generaldist', ['utils', function (utils) {
+      return {
+        restrict: 'E',
+        scope: {
+          data: '=',
+          metadata: '=',
+          initDefaults: '=',
+          readOnly:'='
+        },
+        replace: true,
+        templateUrl: 'components/directives/cache/general/generaldist.html',
+        link: function (scope, element, attrs) {
+          scope.cacheTypes = ['distributed-cache', 'local-cache','replicated-cache', 'invalidation-cache'];
+
+          if (scope.initDefaults){
+            scope.data.type = 'distributed-cache';
+            scope.data.data = {};
+            scope.data.data.mode = 'SYNC';
+            scope.data.data['auto-config'] = scope.metadata['auto-config'].default;
+            scope.data.data['remote-cache'] = scope.metadata['remote-cache'].default;
+            scope.data.data['remote-site'] = scope.metadata['remote-site'].default;
+            scope.data.data['remote-timeout'] = scope.metadata['remote-timeout'].default;
+            scope.data.data['capacity-factor'] = scope.metadata['capacity-factor'].default;
+            scope.data.data['jndi-name'] = scope.metadata['jndi-name'].default;
+            scope.data.data['l1-lifespan'] = scope.metadata['l1-lifespan'].default;
+            scope.data.data['queue-flush-interval'] = scope.metadata['queue-flush-interval'].default;
+            scope.data.data['queue-size'] = scope.metadata['queue-size'].default;
+
+            scope.data.data.batching = scope.metadata.batching.default;
+            scope.data.data.module = scope.metadata.module.default;
+            scope.data.data.owners = scope.metadata.owners.default;
+            scope.data.data.segments = scope.metadata.segments.default;
+            scope.data.data.start = scope.metadata.start.default;
+            scope.data.data.statistics = scope.metadata.statistics.default;
+          }
+
+          scope.$watch("form.cacheName.$valid", function (){
```

Comment:
What if cache name becomes valid first, but then it becomes invalid? Is it still ok to have create enabled? You could add broadcast of disable on invalid name, but the best option would be not to send messages through broadcasting mechanisms, but use directive attributes to share if creation is enabled or not.